[META-5352] - add environment applied api docs

johnchappelledbt opened the pull request:
## What are you changing in this pull request and why?
<!--
Describe your changes and why you're making them. If related to an open issue or a pull request on dbt Core or another repository, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->
This PR:
* Adds api content under `Environments > Applied`.
* Add an "About schema" page with cards to help people navigate to the information they need. [PR](https://github.com/dbt-labs/docs.getdbt.com/pull/7874)
* Rewrites some of the definitions.

## Checklist
<!-- Ensure your PR meets the following items. Feel free to add additiona checklist items to the list if additional checks need to happen before the PR is merged, such as:
- [ ] Needs technical review
- [ ] Change base branch
- [ ] Merge on xyz date
-->
- [ ] The changes in this PR meet the [docs style guide/fundamentals](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) required for all content.
- [ ] Applied the proper versioning rules if the content is for specific dbt version(s): ([version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).).
<!--
PRE-RELEASE VERSION OF dbt (if so, uncomment):
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
<!-- 
ADDING OR REMOVING PAGES (if so, uncomment):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages
-->

johnchappelledbt committed:
add environment applied mdx files, structure, and links

johnchappelledbt commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-snapshots.mdx:
```diff
@@ -0,0 +1,59 @@
+title: "Snapshots object schema"
+sidebar_label: "Snapshots"
+id: "discovery-schema-environment-applied-snapshots"
+parent: "AppliedState"
+target: "snapshots"
+node: "SnapshotAppliedStateNodeConnection"
```
@runleonarun I created a template file, as I got a ways through this, and tried using the `frontmatter` to dynamically populate a few repeated values. I'm having issues with the Vercel build, but I don't have access to it. We can sync up on Monday, to figure out why it isn't working for me.

johnchappelledbt replied:
This didn't work, and caused a render error. I removed it, and it loaded when running locally.

runleonarun replied:
@johnchappelledbt I think we get you a vercel account if it makes your life easier! 

Note that we can ignore the Vercel link generator test. It only creates links for people with vercel accounts (or if a someone with vercel account makes a commit)

johnchappelledbt commented on website/sidebars.js:
```diff
@@ -811,8 +811,32 @@ const sidebarSettings = {
+                      "docs/dbt-cloud-apis/discovery-schema-environment-applied-modelHistoricalRuns",
+                      "docs/dbt-cloud-apis/discovery-schema-environment-applied-lineage",
+                    ],
+                  },
```
I moved the new files under `Applied` in the sidebar, so that Applied is its own toggle. I move the existing `modelHistoricalRuns` under this also.

johnchappelledbt replied:
<img width="1018" height="637" alt="image" src="https://github.com/user-attachments/assets/400ed841-774b-48d4-9a2c-626b2b3c1921" />

johnchappelledbt committed:
fix render issue, remove extra applied sidebar link

johnchappelledbt committed:
adding copy for snapshots docs

johnchappelledbt committed:
remove tool versions file from branch

johnchappelledbt committed:
link out to Applied and Definition docs for more details, from Environment

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied.mdx:
```diff
@@ -1,20 +1,20 @@
+When querying for `applied`, you can use the following arguments.
 
-<!-- <QueryArgsTable queryName="AppliedState" /> -->
+<NodeArgsTable parent="Environment" name="applied" />
```
@johnchappelledbt Should we remove the "Arguments" section? I don't see any arguments listed [on staging](https://docs-getdbt-com-git-jrc-add-environment-applied-e115ec-dbt-labs.vercel.app/docs/dbt-cloud-apis/discovery-schema-environment-applied#arguments). I also don't see any when I[ browse to applied in Apollo](https://metadata.cloud.getdbt.com/graphql?explorerURLState=N4IgJg9gxgrgtgUwHYBcQC4QEcYIE4CeABABQAkyAbgJZ4RKKoCSY6RAQtQOZOoCEASiLAAOkiJEqteoxQlqrIhSQ06DZChZDR4iUQCGAB0MAbagjDCxevSgQBnFPZIAzWo7YBGAAzft1mz0LLgcrXUC9JAgwBDCIiKR9RAD42wcUAFkEFH0wfRy41MCoCBN4JAA5JIQUookAawB3fTwue1q6xOTwooBfDsD+npshiNG9caHekAAaEEoW6n0AIxMHDBAdCREQKTVZFh22ADYAdgAmGYCdtzxHI6IfK6RpuYALBFz8ew3REABBGAoN4QPDUABe+Wo9COO3Ynzw+CIYGWKBgAH0mI0ABznZYAKW8SEahgA8ujvABFJDnc7YgCiAGUACwEZlYfHgggACWOADEKjAXABNXwAJQA0vpjnBwY0AKwAYQA4i5GRUAKo7aa9IA) and look fro arguments.

johnchappelledbt replied:
Yeah, this one doesn't have args. Perhaps we should remove this, for those without args?

runleonarun replied:
@johnchappelledbt Yea let's remove it.

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-model.mdx:
```diff
@@ -0,0 +1,57 @@
+---
+title: "Models object schema"
+sidebar_label: "Model"
```
```suggestion
sidebar_label: "Models"
```

johnchappelledbt replied:
Yeah, I thought that made more sense, but I just pulled that from the Notion doc

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-model.mdx:
```diff
@@ -0,0 +1,57 @@
+---
+title: "Models object schema"
+sidebar_label: "Model"
+id: "discovery-schema-environment-applied-model"
```
```suggestion
id: "discovery-schema-environment-applied-models"
```

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-model.mdx:
```diff
@@ -0,0 +1,57 @@
+---
+title: "Models object schema"
```
should we update the filename to "...applied-models" and not "...model?"

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposure.mdx:
```diff
@@ -0,0 +1,52 @@
+---
+title: "Exposure object schema"
+sidebar_label: "Exposure"
```
```suggestion
sidebar_label: "Exposures"
```

johnchappelledbt replied:
Same here. I got it from the doc, but this certainly makes more sense to me.

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposure.mdx:
```diff
@@ -0,0 +1,52 @@
+---
+title: "Exposure object schema"
```
```suggestion
title: "Exposures object schema"
```

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposure.mdx:
```diff
@@ -0,0 +1,52 @@
+---
```
Should we rename file to "...appled-exposures" instead of "exposure?"

I see "exposures" with an "s" [here](https://metadata.cloud.getdbt.com/graphql?explorerURLState=N4IgJg9gxgrgtgUwHYBcQC4QEcYIE4CeABABQAkyAbgJZ4RKKoCSY6RAQtQOZOoCEASiLAAOkiJEqteoxQlqrIhSQ06DZChZDR4iUQCGAB0MAbagjDCxevSgQBnFPZIAzWo7YBGAAzft1mz0LLgcrXUC9JAgwBDCIiKR9RAD42wcUAFkEFH0wfRy41MCoCBN4JAA5JIQUookAawB3fTwue1q6xOTwooBfDsD+npshiNG9caHekAAaEEoW6n0AIxMHDBAdCREQKTVZFh22ADYAdgAmGYCdtzxHI6IfK6RpuYALBFz8ew3REABBGAoN4QPDUABe+Wo9COO3Ynzw+CIYGWKBgAH0mI0ABznZYAKW8SEahgA8ujvABFJDnc7YgCiAGUACwEZlYfHgggACWOADEKjAXABNXwAJQA0vpjnBwY0AKwAYQA4i5GRUAKo7aa9IA)

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposure.mdx:
```diff
@@ -0,0 +1,52 @@
+---
+title: "Exposure object schema"
+sidebar_label: "Exposure"
+id: "discovery-schema-environment-applied-exposure"
```
```suggestion
id: "discovery-schema-environment-applied-exposures"
```

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposure.mdx:
```diff
@@ -0,0 +1,52 @@
+import { NodeArgsTable, SchemaTable } from "./schema";
+
+<!-- NEED  UPDATE THIS DESCRIPTION -->
+The exposure object allows you to query information about exposures. 
```
```suggestion
The exposures object allows you to query information about exposures. 
```

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied.mdx:
```diff
@@ -1,20 +1,20 @@
+<NodeArgsTable parent="Environment" name="applied" />
 
 ### Example queries
 
```
The following example query isn't structured as nice as the ones you created. Should it be?

johnchappelledbt replied:
That is because there are no args for Applied. I think we could remove this for those that don't have args.

johnchappelledbt replied:
Oh, I think I misread this. I can take a look at the example query again. I only updated a few, with the idea to update after getting the file in place.

runleonarun reviewed (commented):
Hey @johnchappelledbt Just noticed a few things.

johnchappelledbt committed:
rename exposure and model, remove empty args for applied

johnchappelledbt committed:
pluralize model and exposure in sidebars.js

runleonarun committed:
Adding Disco API context (#7874)

## What are you changing in this pull request and why?

Adding more context for disco api and messaging upcoming changes.

## Checklist
<!-- Ensure your PR meets the following items. Feel free to add
additiona checklist items to the list if additional checks need to
happen before the PR is merged, such as:
- [ ] Needs technical review
- [ ] Change base branch
- [ ] Merge on xyz date
-->
- [ ] The changes in this PR meet the [docs style
guide/fundamentals](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
required for all content.
- [ ] Applied the proper versioning rules if the content is for specific
dbt version(s): ([version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
or [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
- [ ] The content in this PR requires a dbt release note, so I added one
to the [release notes
page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).).
<!--
PRE-RELEASE VERSION OF dbt (if so, uncomment):
- [ ] Add a note to the prerelease version [Migration
Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
<!-- 
ADDING OR REMOVING PAGES (if so, uncomment):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links
that point to deleted pages
-->

<!-- vercel-deployment-preview -->
---
🚀 Deployment available! Here are the direct links to the updated files:


-
https://docs-getdbt-com-git-disco-message-dbt-labs.vercel.app/docs/dbt-cloud-apis/discovery-api
-
https://docs-getdbt-com-git-disco-message-dbt-labs.vercel.app/docs/dbt-cloud-apis/discovery-use-cases-and-examples
-
https://docs-getdbt-com-git-disco-message-dbt-labs.vercel.app/docs/dbt-cloud-apis/schema-discovery-about

<!-- end-vercel-deployment-preview -->

runleonarun committed:
environment definition and fix cards

rpourzand committed:
Update schema-discovery-environment-applied-exposures.mdx

adding exposures description

rpourzand committed:
Update schema-discovery-environment-applied-tests.mdx

adding description to tests

rpourzand committed:
Update schema-discovery-environment-applied-tags.mdx

rpourzand committed:
Update schema-discovery-environment-applied-sources.mdx

adding source descriptions

rpourzand committed:
Update schema-discovery-environment-applied-sources.mdx

rpourzand committed:
Update schema-discovery-environment-applied-snapshots.mdx

Snapshot resource description

rpourzand committed:
Update schema-discovery-environment-applied-seeds.mdx

adding seeds

rpourzand committed:
Update schema-discovery-environment-applied-packages.mdx

adding packages

rpourzand committed:
Update schema-discovery-environment-applied-owners.mdx

rpourzand committed:
Update schema-discovery-environment-applied-owners.mdx

updating owners again

rpourzand committed:
Update schema-discovery-environment-applied-models.mdx

adding docs on models

rpourzand committed:
Update schema-discovery-environment-applied-models.mdx

rpourzand committed:
Update schema-discovery-environment-applied-exposure-tile.mdx

data health tile

rpourzand committed:
Update schema-discovery-environment-applied-models.mdx

fixing model description

runleonarun committed:
fixing links

runleonarun commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-definition.mdx:
```diff
@@ -1,12 +1,12 @@
-The environment object allows you to query information about a particular model based on `environmentId`.
+The definition object allows you to query the logical state of a given project node given its most recent manifest generated models.
 
 The [Example queries](#example-queries) illustrate a few fields you can query with this `environment` object. Refer to [Fields](#fields) to view the entire schema, which provides all possible fields you can query.
```
@johnchappelledbt Just noting this:
We can remove [lines 87-92](https://github.com/dbt-labs/docs.getdbt.com/pull/7875/files#diff-0559a17b2978c2e100408c2c35ad504a242e8ec45c0b21b88ec11966e00ed7bbL87-L92) below

> When querying an `environment`, you can use the following fields.
> <SchemaTable nodeName="Environment" />
> When querying the `applied` field of `environment`, you can use the following fields.
> <SchemaTable nodeName="AppliedState" />

johnchappelledbt committed:
add key fields from nodes section, where needed, typo fixes, cleanup

johnchappelledbt committed:
add recommended to Environment sidebar tab

johnchappelledbt committed:
update example queries

johnchappelledbt committed:
update Definition page

rpourzand committed:
Update schema-discovery-environment-applied-resources.mdx

adding description to resources node

rpourzand committed:
Update schema-discovery-environment-applied-lineage.mdx

johnchappelledbt committed:
delete applied files

rpourzand commented on website/docs/docs/dbt-cloud-apis/schema-discovery-environment-applied-exposure-tile.mdx:
```diff
@@ -0,0 +1,47 @@
+
+### Example query
+
+<!-- NEED  UPDATE THIS EXAMPLE -->
```
We can probably remove this if you updated it! example looks good.

johnchappelledbt committed:
remove example comment

runleonarun committed:
conflicts

runleonarun reviewed (approved):
🚢 this :puppy:
